### ISSUE TITLE
Status Check misleading error

### ISSUE THREAD

**Issue by glouischandra** [GoogleContainerTools/skaffold#5424]
### Expected behavior
Clear error message from status check about image access instead of user cancelled

### Actual behavior
Received STATUSCHECK_USER_CANCELLED

### Information

- Skaffold version: v1.19.0
- Operating system: Linux Goobuntu
![image](https://user-images.githubusercontent.com/3677364/108579574-6783e200-72dc-11eb-914e-b2eed1918fc7.png)

- Contents of skaffold.yaml:

```yaml
# To learn more about the skaffold.yaml schema visit
# https://skaffold.dev/docs/references/yaml/
apiVersion: skaffold/v2beta4
kind: Config
build:
  tagPolicy:
    sha256: {}
  # defines where to find the code at build time and where to push the resulting image
  artifacts:
  - image: nodejs-guestbook-backend
    context: src/backend
  - image: nodejs-guestbook-frontend
    context: src/frontend
# defines the Kubernetes manifests to deploy on each run
deploy:
  kubectl:
    manifests:
    - ./kubernetes-manifests/*.service.yaml
    - ./kubernetes-manifests/*.deployment.yaml
profiles:
# use the cloudbuild profile to build images using Google Cloud Build
- name: cloudbuild
  build:
    googleCloudBuild: {}
- name: buildpacks
  build:
    artifacts:
    - image: nodejs-guestbook-backend
      context: src/backend
      buildpack:
        builder: "gcr.io/buildpacks/builder:v1"
    - image: nodejs-guestbook-frontend
      context: src/frontend
      buildpack:
        builder: "gcr.io/buildpacks/builder:v1"

Launch.json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Run on Kubernetes",
            "type": "cloudcode.kubernetes",
            "request": "launch",
            "skaffoldConfig": "${workspaceFolder}/skaffold.yaml",
            "watch": true,
            "cleanUp": true,
            "portForward": true,
            "imageRegistry": "gcr.io/glouischandra-test",
            "debug": [
                {
                    "image": "nodejs-guestbook-frontend",
                    "sourceFileMap": {
                        "${workspaceFolder}/src/frontend": "/frontend"
                    }
                },
                {
                    "image": "nodejs-guestbook-backend",
                    "sourceFileMap": {
                        "${workspaceFolder}/src/backend": "/backend"
                    }
                }
            ]
        }
    ]
}

```

### Steps to reproduce the behavior
I am using Cloud Code AutoGKE run/debug functionality and deploying the sample guestbook nodejs application. 
Created auto-gke clusters and deploying there yields this error:
![image](https://user-images.githubusercontent.com/3677364/108579474-d1e85280-72db-11eb-8cd8-5662708a3495.png)

I can see the warning of image can't be pulled but the final error code seems confusing as it says user_cancelled.

**Comment by tejal29** [comment]
Thanks @glouischandra for opening this issue. 
looks like there were multiple resource checks happening. 
As one of them failed, pending status checks are cancelled. I do have checks in place to report the actual error instead of cancelled.

**Comment by tejal29** [comment]
Is this bug required for auto GKE ?

**Comment by briandealwis** [comment]
This looks like a permission problem @glouischandra.  Are you deploying to a cluster in `glouischandra-test`?  If not, you need to configure registry credentials for the cluster to access your image.  Or you make your registry public.

**Comment by tejal29** [comment]
> This looks like a permission problem @glouischandra. Are you deploying to a cluster in `glouischandra-test`? If not, you need to configure registry credentials for the cluster to access your image. Or you make your registry public.
> 
> (edit: see #4132)

Yes the "image can't be pulled" is the underlying  issue for resource check failure. The "final" status-check failure reason should not be "STATUSCHECK_USER_CANCELLED" but report correctly "STATUSCHECK_IMAGE_PULL_ERR"

**Comment by tejal29** [comment]
Note: not a high priority but good to fix this issue since chances of seeing this bug are high given cluster could run out of capacity and not scale in time when deploying app.

**Comment by aaron-prindle** [comment]
> Note: not a high priority but good to fix this issue since chances of seeing this bug are high given cluster could run out of capacity and not scale in time when deploying app.

@tejal29 this is a p2 bug atm, would it make sense to keep this as p2 and prioritize this in an upcoming milestone or instead bump this to p3?

**Comment by nkubala** [comment]
@aaron-prindle the fix for this issue should be small, so let's keep it around for the next planning cycle and see if we can squeeze it in. if we don't get to it we can drop it down

**Comment by tejal29** [comment]
Is there anyone looking at this issue?

**Comment by briandealwis** [comment]
@quoctruong is seeing this occurring again.

**Comment by gsquared94** [comment]
@tejal29 should this be closed with #6846 ?

**Comment by tejal29** [comment]
sounds good!
